pkg/registry/minion/etcd/etcd_test.go

```diff
@@ -44,7 +44,7 @@ func (fakeConnectionInfoGetter) GetConnectionInfo(host string) (string, uint, ht
 
 func newStorage(t *testing.T) (*REST, *tools.FakeEtcdClient) {
 	etcdStorage, fakeClient := registrytest.NewEtcdStorage(t)
-	storage, _ := NewREST(etcdStorage, fakeConnectionInfoGetter{})
+	storage, _ := NewREST(etcdStorage, false, fakeConnectionInfoGetter{})
```

Comment:
Yes - the pattern that we have in tests (i.e. that we are explicitly calling fakeClient) is causing some races.
This is switched on (and being tested) by integration tests and e2e tests - I though that it would be enough.